File: remote/src/main/java/com/kelsos/mbrc/repository/library/TrackRepository.java
package com.kelsos.mbrc.repository.library;

import com.kelsos.mbrc.dao.TrackDao;
import com.kelsos.mbrc.dao.views.TrackModelView;
import com.kelsos.mbrc.repository.Repository;
import java.util.List;
import rx.Observable;

public interface TrackRepository extends Repository<TrackDao>{
  Observable<List<TrackModelView>> getTracksByAlbumId(long albumId);
  Observable<List<TrackModelView>> getTracks(int offset, int limit);
}

File: mbrc-lib/src/main/kotlin/com/kelsos/mbrc/repository/library/TrackRepository.kt
package com.kelsos.mbrc.repository.library

import com.kelsos.mbrc.dao.TrackDao
import com.kelsos.mbrc.dao.views.TrackModelView
import com.kelsos.mbrc.repository.Repository
import rx.Observable

interface TrackRepository : Repository<TrackDao> {
    fun getTracksByAlbumId(albumId: Long): Observable<List<TrackModelView>>
    fun getTracks(offset: Int, limit: Int): Observable<List<TrackModelView>>
}
